src/schema/v2/artworks.ts

```diff
@@ -26,14 +26,22 @@ const Artworks: GraphQLFieldConfig<void, ResolverContext> = {
     },
   }),
   resolve: (_root, options, { artworksLoader }) => {
-    const { ids, respectParamsOrder } = options
-    const { page, size } = convertConnectionArgsToGravityArgs(options)
+    const { ids: allIDs, respectParamsOrder } = options
+    const { page, size, offset } = convertConnectionArgsToGravityArgs(options)
+
+    // Since `allIDs` can be a very long list, it must be truncated.
+    // We have `page and `size`, and we optimistically assume all of
+    // the artworks are published, so we first paginate the list of `ids`.
+    const ids = allIDs.slice((page - 1) * size - 1, size)
```

Comment:
Out of curiosity, what would it look like to pass these pagination args all the way to gravity? (even if we construct the proper "connection" here). I'm worried about being bitten in the future due to the `unpublished` constraint.

If we think that ☝️ is the right thing to do but would take too long given we're trying to launch, am also OK to write a ticket and make it a fast-follow. But would love your guidance on what you think is feasible/worth doing.

Reply:
> what would it look like to pass these pagination args all the way to gravity?

I'm not sure how that would work. The bug is that the list of `ids` being passed is too long (and must be truncated before showing up to Gravity). So we need to be truncating that anyway. Sending pagination parameters to Gravity would still lead to the reported bug (for long-ish viewing rooms or artist series).

Reply:
Sorry for not explaining myself well! This pattern (of exposing all artwork IDs and "paginating" in metaphysics) feels a little smelly in general for the reasons discussed above. 

I know we discussed the challenge in the past of metaphysics being responsible for the "artworks connection" type, but I'm interested in discussing what other options there might be.